game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/CasualtySelector.java

```diff
@@ -254,6 +260,38 @@ private static void killAmphibiousFirst(final List<Unit> killed, final Collectio
     }
   }
 
+  /** Kill off the air units with the lowest movement */
```

Comment:
1.a)  What are your thoughts to make this method functional? To explain, instead of mutating parameters, the units that should be killed would be returned.

1.b) Otherwise, if not made function, would you mind documenting the side effects of this method, how are the parameters mutated exactly?

2) Documentation, why do we only kill off air units with the lowest movement? A mention that air units are the only units that can move after a battle is the key and that is why we don't do this for land units.

3) Consider extracting the early termination condition to the caller, eg:
```
if(killed.stream().filter(Matches.unitIsAir()).anyMatch()) {
          killLowestMovementAirUnit(...);
}
```

I do think it's a best practice to have as much of the control flow at the top layers and have them call into functional pieces. This means a person can follow the core logic without jumping into additional methods and the additional methods are generally side-effect free and single purpose to accomplishing some task.  <https://github.com/kbilsted/Functional-core-imperative-shell/blob/master/README.md>

4. What were your considerations for (or rather not) unit testing this?  Not worth it, too much of a PITA to create unit objects, or just short on time maybe?